test/integration_test.py

```diff
@@ -559,11 +559,14 @@ def _assert_catalog_complete(self,
 
     def _assert_catalog_empty(self, catalog: CatalogName):
         with self.subTest('catalog_empty', catalog=catalog):
-            hit_counts = {
-                entity_type: len(self._get_entities(catalog, entity_type))
+            entities = {
+                entity_type: self._get_entities(catalog, entity_type)
                 for entity_type in self.entity_types
             }
+            hit_counts = {e: len(h) for e, h in entities.items()}
             log.info('Hit counts are %r', hit_counts)
+            if any(hit_counts.values()):
```

Comment:
This is essentially the worst possible way of doing this. `any(hit_counts.values())` is the same as `sum(hit_counts.values())`. Why do essentially the same check twice (inefficient) but in two different ways (confusing)? Exploit that assertEqual can be passed a message.